src/components/Api.js

```diff
@@ -0,0 +1,105 @@
+import { data } from "autoprefixer";
+
+export class Api {
+  constructor() {
+    // constructor body
+    this._baseUrl = "https://around-api.en.tripleten-services.com/v1";
+    this._headers = {
+      authorization: "2e65c592-5cc5-4cb0-a6bf-23fa612e6f57",
+      "Content-Type": "application/json",
+    };
+  }
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+  //load cards from server
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.